src/include/disk_cache_filesystem.hpp

```diff
@@ -19,6 +20,22 @@ class DiskCacheFileSystem final : public CacheFileSystem {
                       OnDiskCacheConfig cache_directory_p);
   std::string GetName() const override;
 
+  unique_ptr<FileHandle>
+  OpenFile(const string &path, FileOpenFlags flags,
+           optional_ptr<FileOpener> opener = nullptr) override {
+    Value val;
+    if (opener) {
+      if (!local_filesystem->DirectoryExists(
```

Comment:
Existence already checked inside the implementation, so need to double check here.
https://github.com/duckdb/duckdb/blob/2e88160d032453af843b7193ab0d232e42a6f88d/src/common/local_file_system.cpp#L600